### ISSUE TITLE
sortBy not working for optionSource

### ISSUE THREAD

**Issue by akloeber** [dsternlicht/RESTool#186]
According to https://www.npmjs.com/package/restool#option-source it should be possible to sort options in select input by specifying a sort property in `sortBy`. Unfortunately this is not working with the following configuration:

```json
{
  "label": "Item",
  "name": "itemId",
  "required": true,
  "type": "select",
  "optionSource": {
    "url": "/api/items",
    "dataPath": null,
    "displayPath": "name",
    "valuePath": "id",
    "sortBy": ["name"]
  }
}
```
Items returns by `/api/items` look like this:
```json
[
  {"id": 1, "name": "foo"},
  {"id": 2, "name": "bar"},
  {"id": 3, "name": "ack"}
]
```

We are using RESTool 2.2.8.

**Comment by dsternlicht** [comment]
@akloeber weird. As far as I see this being handled:
https://github.com/dsternlicht/RESTool/blob/f6f0b0e4e1995fff6afd6cc781e65946c6021127/src/components/formRow/formRow.comp.tsx#L151

Would you like to fix and create a PR for this?

**Comment by akloeber** [comment]
@dsternlicht The issue with sorting is that at this point the items in the underlying `optionSource` have already been mapped into type `{ display: string, value: string }` so that the source data is not available anymore and `sortBy` would only work for `"display"` and `"value"`.

I have a fix ready but when analyzing the code I also found an inconsistency with sorting of options provided inline via `options`-property. Right now they are also passed into `orderBy()` with second property being always `[]` (i.e. empty array). Effectively that naturally sorts a collection of string literals but keeps order for a collection with complex objects of shape `{ display: string, value: string }`.

What's the desired behavior here?
My personal opinion is that sorting should not be applied for inline options at all as they may be easily arranged in the desired order (which may not be naturally sorted) in the config file directly. For `optionSource` sorting should only be applied if property `sortBy` is defined.

**Comment by dsternlicht** [comment]
@akloeber I agree. For inline options sorting shouldn't apply. It should only apply for `optionSource`.
